src/snowflake/cli/plugins/spcs/services/manager.py

```diff
@@ -83,3 +83,13 @@ def logs(
         return self._execute_schema_query(
             f"call SYSTEM$GET_SERVICE_LOGS('{service_name}', '{instance_id}', '{container_name}', {num_lines});"
         )
+
+    def upgrade_spec(self, service_name: str, spec_path: Path):
+        spec = self._read_yaml(spec_path)
+        query = f"""
+        alter service {service_name}
+        from specification $$
+        {spec}
+        $$
+        """.splitlines()
+        return self._execute_schema_query(strip_empty_lines(query))
```

Comment:
Why we need splitlines  and strip_empty lines? AFAIK we don't use it in other places

Reply:
The intention was to remove white space and blank lines from a multi line query. Previously the query had been constructed in parts so I used strip_empty_lines, however you are right that now a simple .strip() would be enough to remove trailing/leading whitespace. I've made the change.